migration/migration-service/src/main/java/org/wso2/carbon/ei/migration/internal/EIMigrationServiceComponent.java

```diff
@@ -21,46 +21,68 @@
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
 import org.wso2.carbon.ei.migration.migrate.SecureVaultPasswordMigrationClient;
+import org.wso2.carbon.ei.migration.migrate.RegistryTaskDeletionClient;
 import org.wso2.carbon.registry.core.service.RegistryService;
 
-/**
- * @scr.component name="org.wso2.carbon.ei.migration.client" immediate="true"
- * @scr.reference name="registry.service" interface="org.wso2.carbon.registry.core.service.RegistryService"
- * cardinality="1..1" policy="dynamic"  bind="setRegistryService" unbind="unsetRegistryService"
- */
+@Component(
+        name = "org.wso2.carbon.ei.migration.client",
+        immediate = true)
 public class EIMigrationServiceComponent {
 
     private static final Log log = LogFactory.getLog(EIMigrationServiceComponent.class);
 
-    /**
-     * Method to activate bundle.
-     *
-     * @param context OSGi component context.
-     */
+    @Activate
     protected void activate(ComponentContext context) {
-        if (Boolean.valueOf(System.getProperty("migrate"))) {
+        if (Boolean.valueOf(System.getProperty("migrate.esb"))) {
             log.info("WSO2 EI migration bundle is activated");
-            log.info("Initiating WSO2 EI migration");
+            log.info("Initiating WSO2 EI secureVault password migration");
             try {
                 SecureVaultPasswordMigrationClient migrateSecureVaultPasswords = new SecureVaultPasswordMigrationClient();
                 migrateSecureVaultPasswords.migratePasswords();
                 log.info("Successfully completed password migration");
             } catch (Throwable e) {
-                log.error("Error occurred during migration", e);
+                log.error("Error occurred during secure vault password migration", e);
+            }
+
+            log.info("Initiating WSO2 EI Scheduled Message Processor Task Deletion");
+            try {
```

Comment:
Lines from 55- 61 are the same as 62- 71. Shall we move the code into a separate method and refer to it after the if else block since this is common to both cases?

Reply:
Fixed